specification/containerservice/resource-manager/Microsoft.ContainerService/aks/preview/2023-07-02-preview/managedClusters.json

```diff
@@ -0,0 +1,8779 @@
+          "description": "This can either be set to an integer (e.g. '5') or a percentage (e.g. '50%'). If a percentage is specified, it is the percentage of the total agent pool size at the time of the upgrade. For percentages, fractional nodes are rounded up. If not specified, the default is 1. For more information, including best practices, see: https://docs.microsoft.com/azure/aks/upgrade-cluster#customize-node-surge-upgrade"
+        },
+        "drainTimeoutInMinutes": {
+          "type": "integer",
+          "format": "int32",
+          "maximum": 1440,
+          "minimum": 1,
+          "title": "The drain timeout for a node",
+          "description": "The amount of time (in minutes) to wait on eviction of pods and graceful termination per node. This eviction wait time honors waiting on pod disruption budgets. If this time is exceeded, the upgrade fails. If not specified, the default is 30 minutes."
+        }
+      },
+      "description": "Settings for upgrading an agentpool"
+    },
+    "AgentPool": {
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/SubResource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "description": "Properties of an agent pool.",
+          "$ref": "#/definitions/ManagedClusterAgentPoolProfileProperties",
+          "x-ms-client-flatten": true
+        }
+      },
+      "description": "Agent Pool."
+    },
+    "MachineListResult": {
+      "type": "object",
+      "properties": {
+        "value": {
+          "type": "array",
+          "x-ms-identifiers": [
+            "id"
+          ],
+          "items": {
+            "$ref": "#/definitions/Machine"
+          },
+          "description": "The list of Machines in cluster."
+        },
+        "nextLink": {
+          "type": "string",
+          "format": "uri",
+          "description": "The URL to get the next set of machine results.",
+          "readOnly": true
+        }
+      },
+      "description": "The response from the List Machines operation."
+    },
+    "Machine": {
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/SubResource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/MachineProperties",
+          "description": "The properties of the machine"
+        }
+      },
+      "description": "A machine. Contains details about the underlying virtual machine. A machine may be visible here but not in kubectl get nodes; if so it may be because the machine has not been registered with the Kubernetes API Server yet."
+    },
+    "MachineProperties": {
+      "type": "object",
+      "readOnly": true,
+      "description": "The properties of the machine",
+      "properties": {
+        "network": {
+          "$ref": "#/definitions/MachineNetworkProperties"
+        },
+        "resourceId": {
```

Comment:
Please add "format": "arm-id" and "x-ms-arm-id-details", since it's the ID of a VM.

https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-arm-id-details

Reply:
added it